### DISCUSSION TITLE
[7_GPU_and_Specialized_Hardware]同一个thread block里不同thread写同一个位置，是否有race condtition?

### DISCUSSION THREAD

**Discussion by hbsun2113** [mlc-ai/mlc-zh#104] [Q&A] [answered]
在[7_GPU_and_Specialized_Hardware](https://github.com/mlc-ai/notebooks/blob/main/7_GPU_and_Specialized_Hardware.ipynb)发现同一时刻， 不同线程在向shared memory的同一个索引位置写值，请问这里不会出现竞争问题吗？

Window Sum Example:
```
A_shared = sch.cache_read(block_C, read_buffer_index=0, storage_scope="shared")
sch.compute_at(A_shared, i1)
sch.mod.show()
```
![image](https://user-images.githubusercontent.com/27994442/183400443-4d3c96d3-4530-45be-bdb1-9439c27720d1.png)

**Comment by Hzfengsy** [comment, marked as answer]
会出现重复计算问题，但是由于不同线程的结果一致所以是安全的。这也是为什么我们需要cooperative fetch的原因
